src/main/java/com/example/messageserver/service/PriceService.java

```diff
@@ -49,15 +53,33 @@ public BatteryPrice findOneByPrice(double price){
         return priceDALimpl.findOneByPrice(price);
     }
 
-    public BatteryOnlyPriceDto getPrice(){
-        return webClient.get()
-                .uri("/batteryPrice")
-                .retrieve()
-                .bodyToMono(BatteryOnlyPriceDto.class)
-                .block();
+    public BatteryOnlyPriceDto getPrice(String operation){
+            return webClient.get()
+                    .uri(formUri(operation))
+                    .retrieve()
+                    .bodyToMono(BatteryOnlyPriceDto.class)
+                    .block();
+
     }
     public BatteryPrice savePrice(BatteryPriceInput batteryPriceInput){
         BatteryPrice batteryPrice = modelMapper.map(batteryPriceInput, BatteryPrice.class);
         return priceRepository.save(batteryPrice);
     }
+    private String formUri(String operation){
+        switch (operation) {
+
+            case "MEDIUM" :
+                return baseUrl + "/?operation=MEDIUM";
+
+            case "HIGH":
+                return baseUrl + "/operation=HIGH";
+
+            default:
+                return baseUrl + "/?operation=LOW";
+
+        }
+    }
```

Comment:
This will return LOW for everything that isn't MEDIUM or HIGH. Add separate LOW case and default case could log an error and return an empty string.